lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,132 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
```

Comment:
We can change the default for Dragonfly BSD later; rather than trying to change all things pkgng in one shot we can get this one thing taken care of and treat other BSDs as separate issues/PRs.